### PR TITLE
Arc tags

### PR DESCRIPTION
**Pull request by notbrokenjemma**
Allow for custom tags to be added to both the ARC-Seal and ARC-Message-Signature headers.

### CODE REVIEW

**Comment by notbrokenjemma** [comment]
some whitespace issues to be sorted, could do with some tests.

File: lib/Mail/DKIM/ARC/Seal.pm [hunk 1]
```diff
@@ -83,6 +83,7 @@ sub new {
     $self->timestamp(
         defined $prms{'Timestamp'} ? $prms{'Timestamp'} : time() );
     $self->expiration( $prms{'Expiration'} ) if defined $prms{'Expiration'};
+    $self->tags( $prms{'Tags'} ) if defined $prms{'Tags'};
```
Comment (Davis-A) [inline lib/Mail/DKIM/ARC/Seal.pm:86]: Should this be 'SealTags'? In the Pod in `Signer.pm` it states Tags are for the signature and SealTags are for the seal.

File: scripts/arcsign.pl [hunk 2]
```diff
@@ -23,6 +23,8 @@
   [ "chain=s" => "Chain value. 'ar' means pick it up from Authentication-Results header", {default=>"ar"} ],
   [ "key=s" => "File containing private key, without BEGIN or END lines.", {default=>"private.key"} ],
   [ "debug-canonicalization=s" => "Outputs the canonicalized message to the specified file in addition to computing the DKIM signature. This is helpful for debugging canonicalization methods." ],
+ 	[ "extra-tag=s@" => "Extra tags to use in signing" ],
+	[ "extra-seal-tag=s@" => "Extra tags to use in sealing" ],
```
Comment (Davis-A) [inline scripts/arcsign.pl]: Whitespace looks wrong, but that might be githubs ui showing it oddly ???

File: scripts/arcsign.pl [hunk 3]
```diff
@@ -48,6 +50,22 @@
 	binmode STDIN;
 }
 
+my %arc_opt;
```
Comment (Davis-A) [inline scripts/arcsign.pl:51]: Indentation looks really bad again.  Maybe it's just my UI, but it looks really inconsistent.  New editor woes?

image

File: scripts/dkimsign.pl [hunk 4]
```diff
@@ -100,12 +100,11 @@ sub signer_policy
...
-        if ($opt->extra_tag) {
-	        foreach my $extra ($opt->extra_tag->@*)
-	        {
-		        my ($n, $v) = split /=/, $extra, 2;
-	        	$sig->set_tag($n, $v);
-                }
+  if ($opt->extra_tag) {
+    foreach my $extra ($opt->extra_tag->@*) {
+      my ($n, $v) = split /=/, $extra, 2;
+     	$sig->set_tag($n, $v);
```
Comment (Davis-A) [inline scripts/dkimsign.pl:106]: whitespace again?

File: scripts/arcsign.pl (hunk 3 above)
Comment (Davis-A) [reply scripts/arcsign.pl:51]: And then you fixed it in the final commit.  Why do i review one commit at a time 😆

**Review by Davis-A — COMMENTED** [review]
Just the first commit re whether it should be 'SealTags' instead of 'Tags'

File: lib/Mail/DKIM/ARC/Seal.pm (hunk 1 above)
Comment (notbrokenjemma) [reply lib/Mail/DKIM/ARC/Seal.pm:86]: No, that's right, It's Tags internally to each of the ARC::Seal or ARC::MessageSignature classes, but the ARC::Signer class passes them through, SealTags becomes Tags here when creating a new ARC::Seal

$self->{'SealTags'} ? ( Tags => $self->{'SealTags'} )

**Review by Davis-A — APPROVED** [review]

**Pull request merged**